src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -285,10 +332,12 @@ def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
         onnx_input_names = self.multitoken_engine.onnx_input_names_no_cache
         engine_input = self.tokens_to_engine_input(input_tokens, onnx_input_names)
 
-        if inputs.session_id is not None:
-            # if session_id is provided, we need to set it in engines
-            self.engine.session_id = inputs.session_id
-            self.multitoken_engine.session_id = inputs.session_id
+        self._use_session_ids = False
+        if inputs.session_ids is not None:
+            # TODO: This is ugly, we should find a better way to pass
```

Comment:
why not have `NLDecoderEngine` accept an additional kwarg for this? we will only be passing this in when the engine is an `NLDecoderEngine` right?

Reply:
yeah, outsourcing all the `session_id` management to `NLDecoderEngine` that early is a good idea.